src/resources/database.rs

```diff
@@ -1,22 +1,92 @@
+use std::str::FromStr;
+
 use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use sqlx::{any::install_default_drivers, Any, AnyPool, Pool};
 use thiserror::Error;
 
+#[derive(PartialEq, Clone, Copy)]
+pub enum DatabaseKind {
+    Sqlite,
+    Postgres,
+}
+
+#[derive(Debug, Error)]
+#[error("{0}")]
+pub struct DatabaseKindParseError(&'static str);
+
+impl std::str::FromStr for DatabaseKind {
+    type Err = DatabaseKindParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "SQLITE" | "Sqlite" | "sqlite" => Ok(Self::Sqlite),
+            "POSTGRES" | "Postgres" | "postgres" => Ok(Self::Postgres),
```

Comment:
**suggestion**: I would use one of two things here:
1. `to_lowercase` and check; this should only be done during init, so we don't care about the extra allocation to create a `String`
2. Use a crate that provides case-insensitive matches (`unicase` is one I seem to remember)

Reply:
I went with the `to_lowercase` approach.